Zelus oil - CWC gear

Trilbyspaceclone opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Adds a new flask and chemical
Brass flasks are mostly a throwing weapon with a 1 time use for are clock brethren that should deal around 18 harm to a unarmored person.
The new chem Zelus oil, will rapidly heal clock cults but you only get a finite amount. 30u on spawn and need to grind brass to get more.
Splashing Zelus oils onto a metal sheet will make more brass. 1u = 1 brass. - No free oils!
Also added in a fancy message to Cult about a fog purging the darkness in their mind - keeping it more neutral
Changed Clock cult from seeing unholy tenticals, to a fog as to seem more neutral

## Why It's Good For The Game

CWC is a 26~% win rate game, is weaker then Blood cult in most ways.
I believe the problem is a lack of means to heal or stay in combat for any amount of time.
In theory this should help the balance a bit to be more even a playing field 

## Changelog
:cl:
add: Zelus Oil, brass flasks
/:cl:

Trilbyspaceclone committed:
Zelus indeed

Trilbyspaceclone committed:
better wording*

Ghommie commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -206,7 +206,12 @@
+		to_chat(M, "<span class='userdanger'>A fog spreads through your mind, purging the Justiciar's influence!</span>")
+	..()
+
+/datum/reagent/water/holywater/reaction_mob(mob/living/M, method=TOUCH, reac_volume)
```
Merge this with the above, they are litterally the same proc.

Trilbyspaceclone replied:
One is for blood,
One is for oil,
They are kinda different!

Ghommie replied:
It's the same proc. Don't be silly with this copypasting stuff.

Ghommie commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -309,6 +314,44 @@
+		M.AdjustStun(-30, 0)
+		M.AdjustKnockdown(-70, 0)
+		M.adjustStaminaLoss(-15, 0)
+		M.adjustToxLoss(-5, 0)
```
```suggestion
		M.adjustToxLoss(-5, 0, TRUE)
```
slimepeople.

Ghommie commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -309,6 +314,44 @@
+		M.adjustOxyLoss(-3, 0)
+		M.adjustBruteLoss(-3, 0)
+		M.adjustFireLoss(-5, 0)
+	if(iscultist(M))// We REALLY hate blood cult
```
please indent lines all of the below (minus lines 343/344) under another `else`, lest you want clockies to receive 3 points of tox, 2 of oxy and 10 of stamina damage from lines 340 to 342.

Ghommie commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -309,6 +314,44 @@
+		M.AdjustStun(10, 0)
+		M.AdjustKnockdown(20, 0)
+		M.adjustStaminaLoss(15, 0)
+	else	//Were not mean
```
remove this else and un-indent its contents, blood cultists should also die from this chemical, not merely be stunlocked.

Ghommie commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -309,6 +314,44 @@
+//We only get 30u to start with...
+
+/datum/reagent/fuel/holyoil/reaction_obj(obj/O, reac_volume)
+	if(istype(O, /obj/item/stack/sheet/metal))
```
add in the parent call `. = ..()` beforehand, even if it doesn't have much of an effect anyway currently.

Ghommie commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -309,6 +314,44 @@
+	name = "Zelus Oil"
+	id = "holyoil"
+	description = "Oil blessed by a grater being."
+	taste_description = "oils"
```
```suggestion
	taste_description = "metallic oil"
```

Ghommie replied:
or "grease", "brass", or just merely remove this line if you want to make it taste like "gross metal".
"oils" is generic and plural.

Ghommie commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -206,7 +206,12 @@
+
+/datum/reagent/water/holywater/reaction_mob(mob/living/M, method=TOUCH, reac_volume)
+	if(iscultist(M))
+		to_chat(M, "<span class='userdanger'>A fog spreads through your mind, weakening your connect to the viel and purging your mind of Nar-sen's influence</span>")
```
```suggestion
		to_chat(M, "<span class='userdanger'>A fog spreads through your mind, weakening your connection to the veil and purging Nar-sie's influence</span>")
```

yourdoom9898 replied:
"connect" -> "connection"
"viel" -> "veil"
"Nar-sen's" -> "Nar-sie's"

Ghommie commented on code/modules/food_and_drinks/drinks/drinks/bottle.dm:
```diff
@@ -202,6 +202,19 @@
 
+/obj/item/reagent_containers/food/drinks/holyoil
+	name = "flask of zelus oil"
+	desc = "A brass flask of the zelus oil."
```
```suggestion
	desc = "A brass flask of Zelus oil, a viscous fluid with a scenting of brass - this flask may be sipped or thrown."
```
More in line with the flask of unholy water's description.

yourdoom9898 replied:
The word "the" is unnecessary

Ghommie commented on code/modules/food_and_drinks/drinks/drinks/bottle.dm:
```diff
@@ -202,6 +202,19 @@
+	list_reagents = list("holyoil" = 30) //Powerfull
+	volume = 30
+	foodtype = NONE
+	force = 12 //Same as a toolbox
```
this is pointless due to how drinks code totally overrides normal attack().

Trilbyspaceclone replied:
Yep! Thats also intended.

Ghommie commented on code/modules/food_and_drinks/drinks/drinks/bottle.dm:
```diff
@@ -202,6 +202,19 @@
+	force = 12 //Same as a toolbox
+	throwforce = 18
+
+/obj/item/reagent_containers/food/drinks/holyoil/null
```
```suggestion
/obj/item/reagent_containers/food/drinks/bottle/holyoil/empty
```

Trilbyspaceclone replied:
Were not doing bottles for a reason

Ghommie commented on code/modules/food_and_drinks/drinks/drinks/bottle.dm:
```diff
@@ -202,6 +202,19 @@
 	desc = "A flask of holy water...it's been sitting in the Necropolis a while though."
 	list_reagents = list("hell_water" = 100)
 
+/obj/item/reagent_containers/food/drinks/holyoil
```
```suggestion
/obj/item/reagent_containers/food/drinks/bottle/holyoil
```

Ghommie commented on code/modules/antagonists/clockcult/clock_items/clockwork_armor.dm:
```diff
@@ -69,7 +69,7 @@
 	resistance_flags = FIRE_PROOF | ACID_PROOF
 	armor = list("melee" = 60, "bullet" = 70, "laser" = -25, "energy" = 0, "bomb" = 60, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 100)
-	allowed = list(/obj/item/clockwork, /obj/item/clothing/glasses/wraith_spectacles, /obj/item/clothing/glasses/judicial_visor, /obj/item/mmi/posibrain/soul_vessel)
+	allowed = list(/obj/item/clockwork, /obj/item/clothing/glasses/wraith_spectacles, /obj/item/clothing/glasses/judicial_visor, /obj/item/mmi/posibrain/soul_vessel, /obj/item/reagent_containers/food/drinks/holyoil)
```
```suggestion
	allowed = list(/obj/item/clockwork, /obj/item/clothing/glasses/wraith_spectacles, /obj/item/clothing/glasses/judicial_visor, /obj/item/mmi/posibrain/soul_vessel, /obj/item/reagent_containers/food/drinks/bottle/holyoil)
```

Ghommie commented on code/game/objects/items/stacks/sheets/sheet_types.dm:
```diff
@@ -424,6 +424,8 @@ GLOBAL_LIST_INIT(brass_recipes, list ( \
 	new/datum/stack_recipe("receiver - steam vent", /obj/structure/destructible/clockwork/trap/steam_vent, 3, time = 30, one_per_turf = TRUE, on_floor = TRUE, placement_checks = STACK_CHECK_CARDINALS), \
 	new/datum/stack_recipe("receiver - power nullifier", /obj/structure/destructible/clockwork/trap/power_nullifier, 5, time = 20, one_per_turf = TRUE, on_floor = TRUE, placement_checks = STACK_CHECK_CARDINALS), \
+	null,
+	new/datum/stack_recipe("brass flask", /obj/item/reagent_containers/food/drinks/holyoil/null), \
```
```suggestion
	new/datum/stack_recipe("brass flask", /obj/item/reagent_containers/food/drinks/bottle/holyoil/empty), \
```

Ghommie commented on code/game/gamemodes/clock_cult/clock_cult.dm:
```diff
@@ -338,6 +338,10 @@ Credit where due:
 	CLOCKCULTCHANGELOG\
 	</ul>\
 	<hr>\
+	<li><b>Brass Flasks:</b> These contain 30u of Zelus oil, this oil can be used to help stun people, and more son on Blood Cult\
```
```suggestion
	<li><b>Zelus oil:</b> A new reagent. It can be used to heal the faithful to Ratvar, or kill heretics and moreso stun blood cultists,\
```

yourdoom9898 replied:
"more son" -> "more so"

Ghommie commented on code/game/gamemodes/clock_cult/clock_cult.dm:
```diff
@@ -338,6 +338,10 @@ Credit where due:
 	</ul>\
 	<hr>\
+	<li><b>Brass Flasks:</b> These contain 30u of Zelus oil, this oil can be used to help stun people, and more son on Blood Cult\
+	It can also be used to splash onto metal to make brass. Grinding bass will give you some oil back as well.\
```
```suggestion
	or splashed onto metal sheets to make brass. This chemical can be found in minimal quantities by grinding brass sheets.\
```

yourdoom9898 replied:
"bass" -> "brass"

Ghommie commented on code/game/gamemodes/clock_cult/clock_cult.dm:
```diff
@@ -338,6 +338,10 @@ Credit where due:
 	<hr>\
+	<li><b>Brass Flasks:</b> These contain 30u of Zelus oil, this oil can be used to help stun people, and more son on Blood Cult\
+	It can also be used to splash onto metal to make brass. Grinding bass will give you some oil back as well.\
+	You can make more brass flasks to be used as throwing weaponds if you so wish, these will not have any oil in them. They have one throw before braking making them rather useless.\
```
```suggestion
	<li><b>Brass Flasks:</b>Intended to store Zelus Oil in, but can also be used as fragile single use throwing weapons in a pinch! \
```

yourdoom9898 replied:
"throwing weaponds" -> "throwing weapons"

"braking" -> "breaking"

Ghommie commented on code/game/gamemodes/clock_cult/clock_cult.dm:
```diff
@@ -338,6 +338,10 @@ Credit where due:
+	<li><b>Brass Flasks:</b> These contain 30u of Zelus oil, this oil can be used to help stun people, and more son on Blood Cult\
+	It can also be used to splash onto metal to make brass. Grinding bass will give you some oil back as well.\
+	You can make more brass flasks to be used as throwing weaponds if you so wish, these will not have any oil in them. They have one throw before braking making them rather useless.\
+	Lastly the oil can be drank for a bost in combat, gives you stamina as well as heals quickly!\
```
```suggestion
	These are crafted with a single sheet of brass and fit in the Clockwork Cuirass' suit storage.\
```

yourdoom9898 replied:
"bost" -> "boost"

I'd also add the word "it" after the comma, but that's more grammar than anything

Ghommie commented on code/game/gamemodes/clock_cult/clock_cult.dm:
```diff
@@ -275,7 +275,7 @@ Credit where due:
 	belt = /obj/item/storage/belt/utility/servant
 	backpack_contents = list(/obj/item/storage/box/engineer = 1, \
-	/obj/item/clockwork/replica_fabricator = 1, /obj/item/stack/tile/brass/fifty = 1, /obj/item/paper/servant_primer = 1)
+	/obj/item/clockwork/replica_fabricator = 1, /obj/item/stack/tile/brass/fifty = 1, /obj/item/paper/servant_primer = 1, /obj/item/reagent_containers/food/drinks/holyoil = 1)
```
```suggestion
	/obj/item/clockwork/replica_fabricator = 1, /obj/item/stack/tile/brass/fifty = 1, /obj/item/paper/servant_primer = 1,  /obj/item/reagent_containers/food/drinks/bottle/holyoil = 1)
```

Ghommie replied:
This is required for the flask to shatter on hit like any other `/obj/item/reagent_containers/food/drinks/bottle`, don't ignore it.

yourdoom9898 commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -309,6 +314,44 @@
+/datum/reagent/fuel/holyoil		//Its oil
+	name = "Zelus Oil"
+	id = "holyoil"
+	description = "Oil blessed by a grater being."
```
"grater" -> "greater"

Ghommie reviewed (commented):
I may have committed some typos while attempting to fix and improve their sentences, as english is not my main language. If any of you willing reviewers spot some, please tell.

yourdoom9898 commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -309,6 +314,44 @@
+	name = "Zelus Oil"
+	id = "holyoil"
+	description = "Oil blessed by a grater being."
+	taste_description = "oils"
```
"oils" -> "oil" (doesn't need to be a plural)

Trilbyspaceclone replied:
oils as in different types, this one at lest was intented

Trilbyspaceclone committed:
Apply suggestions from code review

Grammer and such

Co-Authored-By: Ghom <42542238+Ghommie@users.noreply.github.com>

Trilbyspaceclone committed:
Update other_reagents.dm

Trilbyspaceclone committed:
this was fine

Trilbyspaceclone committed:
I guess this is better

Trilbyspaceclone committed:
Update other_reagents.dm

Trilbyspaceclone committed:
I dont care anymore, It works in testing to function fine

I dont see a problem is this, kinda would like citation on were Clockies will get added in the last else claim.
